lib/manager.js

```diff
@@ -58,3 +58,20 @@ exports.AssetManager = class AssetManager {
 		return this.manifest.set(originalPath, actualPath, targetDir);
 	}
 };
+
+function generateFingerprint(filepath, data) {
+	let filename = path.basename(filepath);
+	let ext = filename.indexOf(".") === -1 ? "" : "." + filename.split(".").pop();
+	let name = ext.length === 0 ? filename : path.basename(filepath, ext);
+	let hash = generateHash(data);
+	return path.join(path.dirname(filepath), `${name}-${hash}${ext}`);
+}
+
+// exported for testing
+exports.generateFingerprint = generateFingerprint;
```

Comment:
```suggestion
exports._generateFingerprint = generateFingerprint;
```

That makes it explicit; cf. [`_determinePlugins`](https://github.com/faucet-pipeline/faucet-pipeline-core/blob/dae1cd5ec744e386118234afe72a536e158a9015/lib/plugins.js#L31).